client/client-multi/client-c/test/test_protocol.c

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2014-2015 CyberVision, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <time.h>
+#include <unistd.h>
+#include "platform/ext_sha.h"
+
+#include "kaa_logging.h"
+#include "kaa_test.h"
+#include "kaa_context.h"
+#include "kaa_platform_protocol.h"
+#include "kaa_channel_manager.h"
+#include "kaa_platform_utils.h"
+#include "kaa_configuration_manager.h"
+#include "kaa_status.h"
+#include "utilities/kaa_mem.h"
+#include "utilities/kaa_log.h"
+#include "platform/ext_log_storage.h"
+#include "platform/ext_log_upload_strategy.h"
+
+#ifndef KAA_DISABLE_FEATURE_LOGGING
+extern kaa_error_t kaa_status_create(kaa_status_t **kaa_status_p);
+extern kaa_error_t kaa_init(kaa_context_t **kaa_context_p);
+extern kaa_error_t kaa_channel_manager_create(kaa_channel_manager_t **channel_manager_p, kaa_context_t *context);
+extern kaa_error_t kaa_log_collector_create(kaa_log_collector_t ** log_collector_p
+                                          , kaa_status_t *status
+                                          , kaa_channel_manager_t *channel_manager
+                                          , kaa_logger_t *logger);
+
+extern kaa_error_t kaa_platform_protocol_create(kaa_platform_protocol_t **platform_protocol_p, kaa_context_t *context
+                                              , kaa_status_t *status);
+
+extern kaa_error_t kaa_platform_protocol_serialize_client_sync(kaa_platform_protocol_t *self, const kaa_serialize_info_t *info
```

Comment:
This extern is not neeed as the declaration presents in the corresponding header.